### DISCUSSION TITLE
.tsx files not being recognised as Typescript

### DISCUSSION THREAD

**Discussion by marshalljordan1** [github-linguist/linguist#6592] [Classification]
Wrong language detected:

https://github.com/marshalljordan1/git-repo-search

<img width="321" alt="image" src="https://github.com/github-linguist/linguist/assets/96771039/2afddfca-34c2-467c-b352-a7c513642f11">

<img width="1263" alt="image" src="https://github.com/github-linguist/linguist/assets/96771039/6010276f-cd8a-49d8-a640-41bda6bffe79">
<img width="1263" alt="image" src="https://github.com/github-linguist/linguist/assets/96771039/456653d5-c9e6-4c27-b96e-c9dce4bcde88">
<img width="1263" alt="image" src="https://github.com/github-linguist/linguist/assets/96771039/1247eada-a684-4fbe-a282-660bcdb10bda">

**Comment by lildude** [comment]
Nope. Correct language detected. See previous discussion https://github.com/github-linguist/linguist/discussions/5609 which references my explanation in  https://github.com/github-linguist/linguist/issues/5485#issuecomment-888041136

It's also worth pointing out that Search doesn't understand Linguist's language grouping so if you want to find your `.tsx` files by language name, you should use the name TSX.

**Reply by marshalljordan1** [reply]
Thank you for replying! 
I did see that response but I still don't know what I can do to change the % of this:

<img width="321" alt="image" src="https://github.com/github-linguist/linguist/assets/96771039/2afddfca-34c2-467c-b352-a7c513642f11">

It's for a technical interview for my first job and it says I need to write the code in Typescript, which I have. I don't want to be penalised if they see it says 90+% Javascript. Could you help please? I'm not very experienced.

**Reply by lildude** [reply]
You can use an [override](https://github.com/github-linguist/linguist/blob/master/docs/overrides.md) to force the language of those files, but be warned this may result in wrong syntax highlighting.
